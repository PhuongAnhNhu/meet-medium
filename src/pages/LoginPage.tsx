import React from 'react';
import { Card, CardContent, CardHeader } from '@mui/material';
import { Box } from '@mui/system';
import LoginButton from '../components/LoginButton';

const LoginPage = () => {
  return (
    <Box sx={{ height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
      <Card sx={{ maxWidth: 400, height: 300 }}>
        <Box sx={{ display: 'flex', flexFlow: 'column', alignItems: 'center' }}>
          <CardHeader sx={{ bgcolor: 'primary.main' }} title="Willkommen bei MeetMedium" />
        </Box>
        <Box
          sx={{ height: '80%', display: 'flex', flexFlow: 'column', justifyContent: 'center', alignItems: 'center' }}
        >
          <CardContent>
            <LoginButton />
          </CardContent>
        </Box>
      </Card>
    </Box>
  );
};

export default LoginPage;
