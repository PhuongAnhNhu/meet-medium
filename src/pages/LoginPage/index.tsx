import React from 'react';
import { Card, CardContent, CardHeader } from '@mui/material';
import { Box } from '@mui/system';
import LoginButton from '../../components/LoginButton';
import { loginRequest } from '../../authConfig';
import { useMsal } from '@azure/msal-react';

const LoginPage = () => {
  const { instance } = useMsal();

  return (
    <Box sx={{ height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
      <Card sx={{ maxWidth: 400, height: 300 }}>
        <Box sx={{ display: 'flex', flexFlow: 'column', alignItems: 'center' }}>
          <CardHeader sx={{ bgcolor: 'primary.main' }} title="Willkommen bei MeetMedium" />
        </Box>
        <Box
          sx={{ height: '80%', display: 'flex', flexFlow: 'column', justifyContent: 'center', alignItems: 'center' }}
        >
          <CardContent>
            <LoginButton
              onLogin={() =>
                instance.loginRedirect(loginRequest).catch((e) => {
                  console.error(e);
                })
              }
            />
          </CardContent>
        </Box>
      </Card>
    </Box>
  );
};

export default LoginPage;
