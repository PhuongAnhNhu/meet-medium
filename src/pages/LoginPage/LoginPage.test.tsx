import LoginPage from '.';
import { render } from '@testing-library/react';
import { wrapWithStore } from 'tests/wrappers';

describe('LoginPage', () => {
  it('renders', () => {
    const { container } = render(wrapWithStore(<LoginPage />));

    expect(container).toMatchSnapshot();
  });
});
