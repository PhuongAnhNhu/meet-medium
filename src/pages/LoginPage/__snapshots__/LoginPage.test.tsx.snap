// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders 1`] = `
<div>
  <div
    class="MuiBox-root css-1sfjlxw"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1v51ab2-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-728fgs"
      >
        <div
          class="MuiCardHeader-root css-1eo8bax-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Willkommen bei MeetMedium
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-yc89lo"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <img
                alt="ms-logo"
                src="/assets/ms-logo.svg"
              />
            </span>
            Bei Microsoft anmelden
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
